File: frontend/src/types/asset.ts
// ============================================================
// 长期资产库类型定义（二期 M10）
// ============================================================

export interface SOP {
  id: string
  name: string
  category: string | null
  description: string | null
  steps: SOPStep[]
  checklist: string[]
  tags: string[]
  use_count: number
  version: number
  created_at: string
  updated_at: string
}

export interface SOPStep {
  title: string
  description: string
  estimated_time: number | null
}

export interface PromptTemplate {
  id: string
  name: string
  category: string | null
  description: string | null
  role_setting: string | null
  task_description: string | null
  constraints: string | null
  output_format: string | null
  variables: string[]
  use_count: number
  rating: number | null
  created_at: string
  updated_at: string
}

export interface Skill {
  id: string
  name: string
  category: string | null
  description: string | null
  methodology: string | null
  proficiency: 'beginner' | 'intermediate' | 'advanced' | 'expert'
  tags: string[]
  use_count: number
  created_at: string
  updated_at: string
}

export interface ProjectMemory {
  id: string
  project_id: string | null
  name: string
  summary: string | null
  successes: string | null
  failures: string | null
  extracted_assets: string[]
  key_metrics: Record<string, unknown>
  tags: string[]
  created_at: string
  updated_at: string
}

File: backend/app/schemas/asset.py
# ============================================================
# 长期资产库 schema（二期 M10）
# ============================================================
from __future__ import annotations

from datetime import datetime
from pydantic import BaseModel, Field


class SOPOut(BaseModel):
    id: str
    name: str
    category: str | None = None
    description: str | None = None
    steps: list = []
    checklist: list = []
    tags: list = []
    use_count: int = 0
    version: int = 1
    created_at: datetime
    updated_at: datetime


class CreateSOPRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    category: str | None = None
    description: str | None = None
    steps: list = []
    checklist: list = []
    tags: list = []


class PromptTemplateOut(BaseModel):
    id: str
    name: str
    category: str | None = None
    description: str | None = None
    role_setting: str | None = None
    task_description: str | None = None
    constraints: str | None = None
    output_format: str | None = None
    variables: list = []
    use_count: int = 0
    rating: float | None = None
    created_at: datetime
    updated_at: datetime


class CreatePromptTemplateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    category: str | None = None
    description: str | None = None
    role_setting: str | None = None
    task_description: str | None = None
    constraints: str | None = None
    output_format: str | None = None
    variables: list = []


class SkillOut(BaseModel):
    id: str
    name: str
    category: str | None = None
    description: str | None = None
    methodology: str | None = None
    proficiency: str = "beginner"
    tags: list = []
    use_count: int = 0
    created_at: datetime
    updated_at: datetime


class CreateSkillRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    category: str | None = None
    description: str | None = None
    methodology: str | None = None
    proficiency: str = "beginner"
    tags: list = []


class ProjectMemoryOut(BaseModel):
    id: str
    project_id: str | None = None
    name: str
    summary: str | None = None
    successes: str | None = None
    failures: str | None = None
    extracted_assets: list = []
    key_metrics: dict = {}
    tags: list = []
    created_at: datetime
    updated_at: datetime


class CreateProjectMemoryRequest(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    project_id: str | None = None
    summary: str | None = None
    successes: str | None = None
    failures: str | None = None
    tags: list = []
